### PR TITLE
Include Support for Browserify external bundles

### PR DESCRIPTION
**Pull request by Palmaswell**
Add external bundles support to the browserify transform. Appreciate constructive feedback to improve the feature.

### CODE REVIEW

File: source/application/transforms/browserify/index.js [hunk 1]
```diff
@@ -115,6 +115,17 @@ function browserifyTransformFactory (application) {
 			return results;
 		}, {});
 
+		const externalBundles = configuration.external;
+		const externalBundlesNames = (bundles, fn) => {
```
Comment (marionebl) [inline source/application/transforms/browserify/index.js]: `fn` is the bundler instance, please rename to `bundler`

Comment (marionebl) [inline source/application/transforms/browserify/index.js]: Rename function to externalizeModules

File: source/application/transforms/browserify/index.js [hunk 2]
```diff
@@ -115,6 +115,17 @@ function browserifyTransformFactory (application) {
...
+				if (bundles[0] === undefined ) {
```
Comment (marionebl) [inline source/application/transforms/browserify/index.js]: This check should be superfluous

File: source/application/transforms/browserify/index.js [hunk 3]
```diff
@@ -115,6 +115,17 @@ function browserifyTransformFactory (application) {
 			return results;
 		}, {});
 
+		const bundlerConfigNames = ['external', 'exclude', 'ignore'];
+		const applyBundlerConfig = (bundler, method, config) => {
+			if (config[0] === undefined) {
```
Comment (marionebl) [inline source/application/transforms/browserify/index.js]: This check is not needed

File: source/application/transforms/browserify/index.js [hunk 4]
```diff
@@ -115,6 +115,17 @@ function browserifyTransformFactory (application) {
...
+				return;
+			}
+			config.map(file => {
```
Comment (marionebl) [inline source/application/transforms/browserify/index.js]: use .forEach here

File: source/application/transforms/browserify/index.js [hunk 5]
```diff
@@ -134,6 +145,9 @@ function browserifyTransformFactory (application) {
 				}
 			}
 
+			let bundlerConfig = bundlerConfigNames.forEach(bundleOpt => {
```
Comment (marionebl) [inline source/application/transforms/browserify/index.js:145]: Do not assign to bundlerConfig here, seems to be unused.

**Comment by Palmaswell** [comment]
Done ;)

**Comment by marionebl** [comment]
Looks good, could you have a look at this @KnisterPeter, merge and release if everything is fine?

**Comment by marionebl** [comment]
Closing this as discussed, ping me if there are open questions.

**Pull request closed without merging**